hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/IncrementalInputSplits.java

```diff
@@ -356,7 +356,22 @@ private List<MergeOnReadInputSplit> getIncInputSplits(
       LOG.warn("No partitions found for reading under path: " + path);
       return Collections.emptyList();
     }
-    FileStatus[] fileStatuses = WriteProfiles.getFilesFromMetadata(path, hadoopConf, metadataList, metaClient.getTableType());
+    final FileStatus[] fileStatuses;
+    Boolean skipNotExistsFile = conf.getBoolean(FlinkOptions.READ_STREAMING_SKIP_NOT_EXIST_FILE);
+    if (!skipNotExistsFile) {
+      LOG.warn("Allow to skip not exists file, fall back to full table scan.");
```

Comment:
Why fallback to full table scan without checking the file existence?

Reply:
Please see `Change Logs` for details :)